### DISCUSSION TITLE
Custom User Settings Editor

### DISCUSSION THREAD

**Discussion by cohoweb** [OrchardCMS/OrchardCore#11115] [Q&A] [answered]
I've enabled custom user settings and I am able to display the information in the front end.  What I would like to do is allow the users to manage their own custom settings but render them in a custom theme, not the admin theme.  I believe the file I want to leverage is; CustomUserSettings.Edit.cshtml any help is appreciated.

Thanks

**Comment by ns8482e** [comment, marked as answer]
Custom User Settings are stored in IUser's `Properties` 
Read your custom user setting from logged in user's Properties and use ContentDisplayManager to build display or edit.

**Reply by cohoweb** [reply]
Thank you Niraj,  this looks like what I was missing.  I'll follow up with a snippet.

**Reply by cohoweb** [reply]
This works but I'm sure there's a more effective way to leverage Alternates.  
Essentially I copied the Edit method and view from the OrchardCore.Users module (AdminController and Admin/Edit.cshtml) into a custom module:


public async Task<ActionResult> MyProfile(string returnUrl = "")
        {
            var userName = _httpContextAccessor.HttpContext.User.Identity.Name;
            User user = await _userService.GetUserAsync(userName) as User;
            var shape = await _userDisplayManager.BuildEditorAsync(user, updater: _updateModelAccessor.ModelUpdater, isNew: false);
            ViewData["ReturnUrl"] = returnUrl;
            return View(shape);            
        }


and the view:


@model dynamic
        <form asp-action="Edit" asp-area="OrchardCore.Users" asp-controller="Admin" method="post" 
              asp-route-returnUrl="@ViewData["returnUrl"]" enctype="multipart/form-data" class="no-multisubmit">
            @Html.ValidationSummary()
            @await DisplayAsync(Model)
        </form>
